### ISSUE TITLE
validate_token - resource_name - undefined method `name' for nil:NilClass

### ISSUE THREAD

**Issue by charlesdg** [lynndylanhurley/devise_token_auth#480]
Still have the same problem as #348 with a different configuration when I call `/api_v2/auth/validate_token`.

I'm using `ng-token-auth` and here are my routes declaration:

```ruby
namespace :api_v2, defaults: { format: :json } do
    mount_devise_token_auth_for "Identity",
                                at:          "auth",
                                controllers: {
                                  omniauth_callbacks: "api_v2/omniauth_callbacks",
                                  sessions:           "api_v2/sessions",
                                  registrations:      "api_v2/registrations",
                                  confirmations:      "api_v2/confirmations",
                                  passwords:          "api_v2/passwords"
                                }
```

As my app is not an api only app, I need devise_token_auth for this namespace only.

In the `DeviseTokenAuth::ApplicationController`, the controller calls
```ruby
mapping = Devise.mappings[resource_name] || Devise.mappings.values.first
``` 
but `resource_name` is not ok.

Check the trace below. 

```ruby
Started GET "/api_v2/auth/validate_token" for ::1 at 2015-12-21 16:28:00 +0100
Processing by DeviseTokenAuth::TokenValidationsController#validate_token as HTML
request.headers['CF-Connecting-IP']: 
request.remote_ip: ::1
Completed 500 Internal Server Error in 2ms (ActiveRecord: 0.0ms)

NoMethodError - undefined method 'name' for nil:NilClass:
  devise (3.5.3) app/controllers/devise_controller.rb:38:in 'resource_name'
  devise_token_auth (0.1.37.beta4) app/controllers/devise_token_auth/application_controller.rb:11:in 'resource_class'
  devise_token_auth (0.1.37.beta4) app/controllers/devise_token_auth/concerns/set_user_by_token.rb:21:in 'set_user_by_token'
```

If you need more information, feel free to ask me.

**Comment by paulosoares86** [comment]
hey @charlesdg, could you please try my branch bugfix/resource_name (https://github.com/paulosoares86/devise_token_auth/tree/bugfix/resource_name) ? Please tell me if it fixes your issue. If it does, I'll submit a PR.

**Comment by paulosoares86** [comment]
Hey @charlesdg, could you please do the same test with my branch ? Just change the the following line  in your Gemfile

`gem 'devise_token_auth'` 

by 

`gem 'devise_token_auth', :git => 'git://github.com/paulosoares86/devise_token_auth.git', branch: 'bugfix/resource_name'`

If this fixes the issue I will submit a PR. 
Thanks!

**Comment by charlesdg** [comment]
@paulosoares86, yep it fixed the issue at my side. Looking forward for your pull request! Thanks!

**Comment by paulosoares86** [comment]
Hey @booleanbetrayal, I think I've fixed this issue! The method `resource_name` was throwing an exception because `devise_scope` inside `mount_devise_token_auth_for` were called with a wrong argument (a nonexistent Devise mapping). This error can pass without been noticed if there is already a mapping without namespace for the same resource. 

Could you please take a look at the PR ? Please tell me if have any suggestions.

Thanks!

**Comment by booleanbetrayal** [comment]
This should be fixed now as of #484
